kvs/include/utils/server_utils.hpp

```diff
@@ -97,64 +129,142 @@ class EBSSerializer : public Serializer {
     }
     return res;
   }
 
-  bool put(const Key& key, const std::string& value,
-           const unsigned& timestamp) {
-    bool replaced = false;
-    TimestampValuePair<std::string> p =
-        TimestampValuePair<std::string>(timestamp, value);
+  unsigned put(const Key& key, const std::string& serialized) {
+    LWWValue input_value;
+    input_value.ParseFromString(serialized);
 
-    DataValue original_value;
-    DataValue new_value;
+    LWWValue original_value;
 
     std::string fname = ebs_root_ + "ebs_" + std::to_string(tid_) + "/" + key;
     std::fstream input(fname, std::ios::in | std::ios::binary);
 
     if (!input) {  // in this case, this key has never been seen before, so we
                    // attempt to create a new file for it
-      replaced = true;
-      new_value.set_timestamp(timestamp);
-      new_value.set_value(value);
 
       // ios::trunc means that we overwrite the existing file
       std::fstream output(fname,
                           std::ios::out | std::ios::trunc | std::ios::binary);
-      if (!new_value.SerializeToOstream(&output)) {
+      if (!input_value.SerializeToOstream(&output)) {
         std::cerr << "Failed to write payload." << std::endl;
       }
+      return output.tellp();
     } else if (!original_value.ParseFromIstream(
                    &input)) {  // if we have seen the key before, attempt to
                                // parse what was there before
       std::cerr << "Failed to parse payload." << std::endl;
+      return 0;
     } else {
-      // get the existing value that we have and merge
-      ReadCommittedPairLattice<std::string> l =
-          ReadCommittedPairLattice<std::string>(TimestampValuePair<std::string>(
-              original_value.timestamp(), original_value.value()));
-      replaced = l.merge(p);
-
-      if (replaced) {
-        // set the payload's data to the merged values of the value and
-        // timestamp
-        new_value.set_timestamp(l.reveal().timestamp);
-        new_value.set_value(l.reveal().value);
-
-        // write out the new payload.
-        std::fstream output(fname,
-                            std::ios::out | std::ios::trunc | std::ios::binary);
-
-        if (!new_value.SerializeToOstream(&output)) {
+      std::fstream output(fname,
+                          std::ios::out | std::ios::trunc | std::ios::binary);
+      if (input_value.timestamp() >= original_value.timestamp()) {
+        if (!input_value.SerializeToOstream(&output)) {
           std::cerr << "Failed to write payload" << std::endl;
         }
       }
+      return output.tellp();
     }
+  }
+
+  void remove(const Key& key) {
+    std::string fname = ebs_root_ + "ebs_" + std::to_string(tid_) + "/" + key;
 
-    return replaced;
+    if (std::remove(fname.c_str()) != 0) {
+      std::cerr << "Error deleting file" << std::endl;
+    }
+  }
+};
+
+class EBSSetSerializer : public Serializer {
```

Comment:
At a first glance, it looks like the logic for serializing the data is the same across both the memory and EBS serializers mod the persistence model. If I'm not wrong, can we unify the serialization logic and just leave the persistence models for EBS/Memory unified?